crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -51,6 +51,14 @@ def create_proposal
   end
 
   def validate_proposal_after_save proposal
+    proposal["attributes"]["provisioner"]["packages"].each do |platform, packages|
+      packages.each do |package|
+        unless package =~ /(^\w[\w\+\.<=>-]+$)/
```

Comment:
I'm not an regex specialist, but LGTM. There are anyway no packages below 2 chars, so that sounds ok to me.